pyccel/codegen/printing/ccode.py

```diff
@@ -412,12 +412,14 @@ def arrayFill(self, expr):
         lhs = expr.lhs
         code_init = ''
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
-
+        dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
+        dtype = dtype[3:]
+ 
         if rhs.fill_value is not None:
             if isinstance(rhs.fill_value, Literal):
-                code_init += 'array_fill(({0}){1}, {2});\n'.format(declare_dtype, self._print(rhs.fill_value), self._print(lhs))
+                code_init += 'array_fill_{0}(({1}){2}, {3});\n'.format(dtype, declare_dtype, self._print(rhs.fill_value), self._print(lhs))
             else:
-                code_init += 'array_fill({0}, {1});\n'.format(self._print(rhs.fill_value), self._print(lhs))
+                code_init += 'array_fill_{0}({1}, {2});\n'.format(dtype, self._print(rhs.fill_value), self._print(lhs))
```

Comment:
Is there a difference between `declare_dtype` and `dtype`?

Reply:
`declare_dtype`: have the type name as written in C, we use them in the declaration.
`dtype` : has the name used in `ndarrays` funcs to pass the types and in the union to define the type of the array. they are written as `nd_`+`abv of the type`.